include/ignition/math/Helpers.hh

```diff
@@ -830,6 +831,131 @@ namespace ignition
+    /// string's time value
+    inline std::chrono::steady_clock::time_point stringToTimePoint(
+        const std::string &_timeString)
+    {
+      std::chrono::steady_clock::time_point timePoint =
+        math::secNsecToTimePoint(-1, 0);
+
+      if (_timeString.empty())
+        return timePoint;
+
+      // The following regex takes a time string in the general format of
+      // "dd hh:mm:ss.nnn" where n is milliseconds, if just one number is
+      // provided, it is assumed to be seconds
+      std::regex time_regex(
+          "^([0-9]+ ){0,1}"                       // day:
+                                                  // Any positive integer
+
+          "(?:([1-9]:|[0-1][0-9]:|2[0-3]:){0,1}"  // hour:
+                                                  // 1 - 9:
+                                                  // 01 - 19:
+                                                  // 20 - 23:
+
+          "([0-9]:|[0-5][0-9]:)){0,1}"            // minute:
+                                                  // 0 - 9:
+                                                  // 00 - 59:
+
+          "(?:([0-9]|[0-5][0-9]){0,1}"            // second:
+                                                  // 0 - 9
+                                                  // 00 - 59
+
+          "(\\.[0-9]{1,3}){0,1})$");              // millisecond:
+                                                  // .0 - .9
+                                                  // .00 - .99
+                                                  // .000 - 0.999
+      std::smatch matches;
+
+      // `matches` should always be a size of 6 as there are 6 matching
+      // groups in the regex.
+      // 1. The whole regex
+      // 2. The days
+      // 3. The hours
+      // 4. The minutes
+      // 5. The seconds
+      // 6. The milliseconds
+      // We can also index them as such below.
+      // Note that the space will remain in the day match, the colon
+      // will remain in the hour and minute matches, and the period will
+      // remain in the millisecond match
+      if (!std::regex_search(_timeString, matches, time_regex) ||
+          matches.size() != 6)
+        return timePoint;
+
+      uint64_t numberDays = 0;
+      uint64_t numberHours = 0;
+      uint64_t numberMinutes = 0;
+      uint64_t numberSeconds = 0;
+      uint64_t numberMilliseconds = 0;
+      std::string dayString = matches[1];
+      std::string hourString = matches[2];
+      std::string minuteString = matches[3];
+      std::string secondString = matches[4];
+      std::string millisecondString = matches[5];
+
+      // Days are the only unbounded number, so check first to see if stoi
+      // runs successfully
+      if (!dayString.empty())
+      {
+        // Erase the space
+        dayString.erase(dayString.length() - 1);
+        try
+        {
+          numberDays = std::stoi(dayString);
+        }
+        catch (const std::out_of_range &oor)
```

Comment:
another windows warning found by sdformat:

~~~
'oor': unreferenced local variable
~~~

* https://build.osrfoundation.org/job/sdformat-ci-sdformat9-windows7-amd64/37/msbuild/#issuesContent

Reply:
I think this would be easy enough to fix by just deleting `oor`:

~~~
catch (const std::out_of_range &)
~~~

Reply:
Thanks for pointing this out, I'm creating a PR now